engine/consensus/matching/engine.go

```diff
@@ -1140,16 +1163,7 @@ func (e *Engine) requestPendingReceipts() (int, uint64, error) {
 	// heights would stop the sealing.
 	missingBlocksOrderedByHeight := make([]flow.Identifier, 0, e.maxResultsToRequest)
 
-	// turn mempool into Lookup table: BlockID -> Result
-	knownResultForBlock := make(map[flow.Identifier]struct{})
-	for _, r := range e.incorporatedResults.All() {
-		knownResultForBlock[r.Result.BlockID] = struct{}{}
-	}
-	for _, s := range e.seals.All() {
-		knownResultForBlock[s.Seal.BlockID] = struct{}{}
-	}
```

Comment:
Double check here. The reason we remove is that: If we have a seal, we must have enough receipts in storage that could skip requesting.

Reply:
You are right, this allows us to shortcut all the logic for figuring out if we still need to request receipts:
* if we already have a candidate seal, the answer is obviously no

As the logic for figuring out if we still need to request receipts involves quite a few steps (including a direct database read), I put the shortcut back in. (just renamed the map)